File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>
typedef long long ll;
using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<ll,vector<ll>,greater<ll>>pq;
    for(auto x : scoville) pq.push(x);
    
    while(pq.size() > 1 && pq.top() < K){
        answer++;
        ll a = pq.top(); pq.pop();
        ll b = pq.top(); pq.pop();
        ll mix = a + 2 * b;
        pq.push(mix);
    }
    if(pq.top() >= K) return answer;
    else return -1;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Long> pq = new PriorityQueue<>();
        for(int scov : scoville) pq.add((long)scov);
        while(pq.peek() < K && pq.size() > 1){
            long a = pq.poll();
            long b = pq.poll() * 2;
            pq.add(a + b);
            answer++;
        }
        if(pq.peek() < K) return -1;
        return answer;
    }
}